src/pages/admin/donate/components/bank-details-currencies/bank-details-factory/BankDetailsFactory.ts

```diff
@@ -0,0 +1,152 @@
+import { DONATE_TEXT } from '../../../../../../const/admin/donate';
+import { BANK_DETAILS_VALIDATION_FUNCTIONS } from '../../../../../../validation/admin/bank-details-schema/bank-details-schema';
+import { createGenericForm, GenericFormField } from '../../generic-form/GenericForm';
+
+export interface BaseBankDetails {
+    id?: number;
+    name: string;
+    receiver: string;
+    iban: string;
+}
+
+export interface ForeignBankDetails extends BaseBankDetails {
+    swift: string;
+    address: string;
+    correspondentBanks: CorrespondentBankDetails[];
+}
+
+export interface UahBankDetails extends BaseBankDetails {
+    edrpou: string;
+    paymentPurpose: string;
+}
+
+export interface CorrespondentBankDetails {
+    id?: number;
+    name: string;
+    swift: string;
+    account: string;
+    iban?: string;
+}
+
+const baseFields: GenericFormField<BaseBankDetails>[] = [
+    {
+        name: 'name',
+        isTitle: true,
+        validate: (val) => BANK_DETAILS_VALIDATION_FUNCTIONS.validateName(String(val ?? '')),
```

Comment:
use DRY function `withNullCheck` instead
```suggestion
        validate: withNullCheck(BANK_DETAILS_VALIDATION_FUNCTIONS.validateName),
```